langchain/src/chains/llm_chain.ts

```diff
@@ -14,26 +22,56 @@ import {
   Callbacks,
 } from "../callbacks/manager.js";
 import { NoOpOutputParser } from "../output_parsers/noop.js";
+import {
+  Runnable,
+  RunnableBinding,
+  RunnableWithFallbacks,
+} from "../schema/runnable/base.js";
+import { RunnableBranch } from "../schema/runnable/branch.js";
 
+type LLMType =
+  | BaseLanguageModel
+  | Runnable<BaseLanguageModelInput, string>
+  | Runnable<BaseLanguageModelInput, BaseMessage>;
+
+type CallOptionsIfAvailable<T> = T extends { CallOptions: infer CO } ? CO : any;
 /**
  * Interface for the input parameters of the LLMChain class.
  */
 export interface LLMChainInput<
   T extends string | object = string,
-  L extends BaseLanguageModel = BaseLanguageModel
+  Model extends LLMType = LLMType
 > extends ChainInputs {
   /** Prompt object to use */
   prompt: BasePromptTemplate;
   /** LLM Wrapper to use */
-  llm: L;
+  llm: Model;
   /** Kwargs to pass to LLM */
-  llmKwargs?: this["llm"]["CallOptions"];
+  llmKwargs?: CallOptionsIfAvailable<Model>;
   /** OutputParser to use */
   outputParser?: BaseLLMOutputParser<T>;
   /** Key to use for output, defaults to `text` */
   outputKey?: string;
 }
 
+function _getLanguageModel(llmLike: Runnable): BaseLanguageModel {
+  // eslint-disable-next-line no-instanceof/no-instanceof
+  if (llmLike instanceof BaseLanguageModel) {
```

Comment:
Hey sorry one more thing - worried about regression since `instanceof` can get messed up by bundlers. Can we use a type guard here? I'm fine with the other instanceof checks